thavalon-webapp/src/components/Logout.tsx

```diff
@@ -1,13 +1,22 @@
 import React, { useEffect } from 'react';
 import { Redirect } from 'react-router-dom';
-import { log_out } from '../utils/account_utils';
+import AccountManager, { HttpResponse } from '../utils/accountManager';
 
 type LogoutProps = {
     setLoggedIn: any
 };
 
 function Logout(props: LogoutProps) {
-    useEffect(() => props.setLoggedIn(!log_out()));
+    useEffect(() => {
+        const accountManager: AccountManager = AccountManager.getInstance();
+        accountManager.logoutUser().then((httpResponse: HttpResponse) => {
+            if (httpResponse.result) {
+                props.setLoggedIn(false);
+            } else {
+                console.log("Failed to log out: " + httpResponse.message);

```

Comment:
We might want error handling here. That being said, if logout fails, something has so horribly broken on the server side that the user might as well just close the browser and hope they are logged out.

Reply:
Not sure how we'd display the error to the user at all - i'd rather not let them know things are horribly broken